src/algorithm/minkowski/gjk/mod.rs

```diff
@@ -107,11 +108,78 @@ where
                     return Some(simplex);
                 }
             }
-            i += 1;
-            if i >= MAX_ITERATIONS {
+        }
+
+        None
+    }
+
+    /// Compute the distance between the given primitives.
+    ///
+    /// ## Parameters:
+    ///
+    /// - `left`: left primitive
+    /// - `left_transform`: model-to-world-transform for the left primitive
+    /// - `right`: right primitive,
+    /// - `right_transform`: model-to-world-transform for the right primitive
+    ///
+    /// ## Returns:
+    ///
+    /// Will optionally return the distance between the objects. Will return None, if the objects
+    /// are colliding.
+    pub fn distance<P, PL, PR, TL, TR>(
+        &self,
+        left: &PL,
+        left_transform: &TL,
+        right: &PR,
+        right_transform: &TR,
+    ) -> Option<P::Scalar>
+    where
+        P: EuclideanSpace,
+        P::Scalar: BaseFloat,
+        PL: SupportFunction<Point = P>,
+        PR: SupportFunction<Point = P>,
+        SP: SimplexProcessor<Point = P>,
+        P::Diff: Neg<Output = P::Diff> + InnerSpace,
+        TL: Transform<P>,
+        TR: Transform<P>,
+    {
+        let zero = P::Diff::zero();
+        let tolerance: P::Scalar = NumCast::from(GJK_DISTANCE_TOLERANCE).unwrap();
+        let right_pos = right_transform.transform_point(P::origin());
+        let left_pos = left_transform.transform_point(P::origin());
+        let mut d = right_pos - left_pos;
+        let mut simplex: Vec<SupportPoint<P>> = Vec::default();
+        simplex.push(SupportPoint::from_minkowski(
```

Comment:
nit: could be `for &d in &[right_pos - left_pos, left_pos - right_pos] {...}`

Reply:
Yeah, this is a remnant of my copying code from the other algorithm, where d was reused later. For this algorithm it's not, so I'll change it.